stratum/hal/bin/barefoot/docker/Dockerfile.builder

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2019-present Open Networking Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM stratumproject/build:build as builder
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python \
+    python-yaml \
+    python-pip \
+    libssl-dev \
+    libelf-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy stratum source code, include SDE and Linux headers tarball
+ADD . /stratum
```

Comment:
we don't need to copy everything now since we are not building the stratum in this container

Reply:
I'll try if copying only two tarballs works

Reply:
Several updates:
- SDE and header tarballs now can be placed anywhere on the build node and the script takes paths to the tarballs as arguments. When building SDE and kernel modules, the tarballs are copied to the directory where the Dockerfile is located and docker build also starts form there. Internally docker copies ONLY the tarballs to the builder container. The copied tarballs are then removed after the build.
- When building the runtime image, ONLY stratum code is copied to the builder container.
- "-i" options (to keep intermediate builder image) is removed so now it always saves the builder image locally. This is added to the README as a note. Let me know if you think it's better to not save the builder image by default.
- "-b" option (to specify a builder image and skip SDE/kernel modules building) is also removed for simplicity since when the tarballs stay the same and the corresponding builder image exists, building SDE/kernel modules only takes a second. So we don't really need to skip the build. Let me know if this makes sense.